apiVersion: openapi/v1alpha1
kind: Radar
metadata:
    name: "19779978"
    labels:
        datastore_id: "5734813541597184"
data:
    classification: Crash
    created: "2015-02-10T12:07:15.930499Z"
    description: "Summary:\r\nWhen using a closed range in a switch case, and then end of the range is the maximum value for the variable type (e.g.: 0xFF for UInt8, 0xFFFF for UIn16, etc...), the application will crash at runtime when reaching the switch\r\n\r\nSteps to Reproduce:\r\nSample code\r\n\r\nlet a: UInt8 = 0 // Any number here, doesn't matter\r\nswitch a {\r\ncase 0xF0...0xFF: // Any start value, doesn't matter\r\n    println(\"a\")\r\ndefault:\r\n    break\r\n}\r\n\r\nExpected Results:\r\nSwitch is executed correctly\r\n\r\nActual Results:\r\nConsole output:\r\n\r\nfatal error: Range end index has no valid successor\r\n\r\nException stack:\r\n\r\n* thread #1: tid = 0x29ac1, 0x000000010589e0de libswiftCore.dylib`function signature specialization <Arg[0] = Exploded, Arg[1] = Exploded, Arg[2] = Dead, Arg[3] = Dead> of Swift._fatalErrorMessage (Swift.StaticString, Swift.StaticString, Swift.StaticString, Swift.UInt) -> () + 158, queue = 'com.apple.main-thread', stop reason = EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)\r\n  * frame #0: 0x000000010589e0de libswiftCore.dylib`function signature specialization <Arg[0] = Exploded, Arg[1] = Exploded, Arg[2] = Dead, Arg[3] = Dead> of Swift._fatalErrorMessage (Swift.StaticString, Swift.StaticString, Swift.StaticString, Swift.UInt) -> () + 158\r\n    frame #1: 0x0000000105dfc4d3 $__lldb_expr332`main + 1235 at playground332.swift:9\r\n    frame #2: 0x0000000105673230 MyPlayground`_mh_execute_header + 25136\r\n    frame #3: 0x00000001056757f1 MyPlayground`reabstraction thunk helper from @callee_owned () -> (@unowned ()) to @callee_owned (@in ()) -> (@out ()) + 17\r\n    frame #4: 0x0000000105674a51 MyPlayground`partial apply forwarder for reabstraction thunk helper from @callee_owned () -> (@unowned ()) to @callee_owned (@in ()) -> (@out ()) + 81\r\n    frame #5: 0x0000000105675870 MyPlayground`reabstraction thunk helper from @callee_owned (@in ()) -> (@out ()) to @callee_owned () -> (@unowned ()) + 32\r\n    frame #6: 0x00000001056758a7 MyPlayground`reabstraction thunk helper from @callee_owned () -> (@unowned ()) to @callee_unowned @objc_block () -> (@unowned ()) + 39\r\n    frame #7: 0x00007fff8d5992ac CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12\r\n    frame #8: 0x00007fff8d58b3b5 CoreFoundation`__CFRunLoopDoBlocks + 341\r\n    frame #9: 0x00007fff8d58ab6e CoreFoundation`__CFRunLoopRun + 910\r\n    frame #10: 0x00007fff8d58a598 CoreFoundation`CFRunLoopRunSpecific + 296\r\n    frame #11: 0x00007fff90b2cc4f HIToolbox`RunCurrentEventLoopInMode + 235\r\n    frame #12: 0x00007fff90b2c8ce HIToolbox`ReceiveNextEventCommon + 179\r\n    frame #13: 0x00007fff90b2c80b HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 71\r\n    frame #14: 0x00007fff8c776a91 AppKit`_DPSNextEvent + 964\r\n    frame #15: 0x00007fff8c776090 AppKit`-[NSApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 194\r\n    frame #16: 0x00007fff8c76bd93 AppKit`-[NSApplication run] + 594\r\n    frame #17: 0x00000001056735a5 MyPlayground`main + 885\r\n    frame #18: 0x00007fff8e5da5c9 libdyld.dylib`start + 1"
    email: exsephiroth87@gmail.com
    modified: "2015-02-10T12:07:15.930748Z"
    number: "19779978"
    number_intvalue: 19779978
    originated: 10/02/2015
    parent_number: '&{NULL_VALUE}'
    product: Developer Tools
    product_version: Xcode 6.3 beta1 (6D520o) / Swift 1.2
    reproducible: Alway
    resolved: ""
    status: Open
    title: 'Swift 1.2: runtime exception when closed range includes max value for type'
